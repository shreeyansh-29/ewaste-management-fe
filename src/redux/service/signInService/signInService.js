import {SIGN_IN} from "../../../container/constant/constant";
import api from "../../../core/utilities/httpProvider";
export const signInService = (data) => {
  return api.post(SIGN_IN, {
    email: data.payload.email,
    password: data.payload.password,
  });
};
