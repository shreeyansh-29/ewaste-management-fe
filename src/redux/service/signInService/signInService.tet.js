// /* eslint-disable no-undef */
// import {signInService} from "./signInService";
// let fetch = require("node-fetch");

// fetch = jest.fn(() => Promise.resolve());

// describe("test signInService", () => {
//   it("test fetch", () => {
//     const payload = {email: "customer1@gmail.com", password: "123456"};
//     const api = "http://localhost:8083/signin";
//     signInService(payload);
//     const {email} = payload.email || {};
//     const {password} = payload.password || {};
//     expect(fetch.post).toHaveBeenCalledWith(api, {
//       email: email,
//       password: password,
//     });
//   });
// });
