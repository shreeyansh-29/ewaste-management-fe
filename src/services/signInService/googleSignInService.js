import api from "../../core/utilities/httpProvider";

export const googleSignInService = (data) => {
  return api.post(`http://localhost:8083/signin/google?email=${data.payload}`);
};
