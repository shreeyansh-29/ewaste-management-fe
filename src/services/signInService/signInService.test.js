/* eslint-disable no-undef */
import React from "react";
import {signInService} from "./signInService";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({adapter: new Adapter()});

it("testing signInService", () => {
  const data = {payload: {email: "customer1@gmail.com", password: "123456"}};
  const Service = signInService;
  const wrapper = shallow(<Service {...data} />);
  expect.assertions(1);
  expect(toJson(wrapper)).toMatchSnapshot();
});
