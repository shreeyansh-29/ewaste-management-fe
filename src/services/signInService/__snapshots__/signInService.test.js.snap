// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing signInService 1`] = `<Component />`;
